plugins/process/src/main/java/org/radargun/service/JavaProcessService.java

```diff
@@ -152,12 +158,27 @@ public String getJavaPIDs() {
          }
          envs.put(JAVA_HOME, java);
       }
-      if (javaOpts != null) {
+      overwriteJavaOpts(envs, javaOpts);
+
+      // the default implementation is using setArgs. I would like to avoid doing this refactor
+      if (flightRecorder.isEnabled()) {
+         List<String> flightRecorderArgs = new ArrayList<>(1);
+         flightRecorder.setArgs(flightRecorderArgs);
+         StringBuilder flightRecorderArgBuilder = new StringBuilder();
+         flightRecorderArgs.forEach(arg -> {
+            flightRecorderArgBuilder.append(arg).append(" ");
+         });
+         overwriteJavaOpts(envs, flightRecorderArgBuilder.toString());
```

Comment:
This looks like adding flight recorder overrides `javaOpts` parameter. Please add it instead (don't worry about duplicate args here). The `javaOpts` directly overwrites `JAVA_OPTS` env var because it that's just a different way to do the same thing.